src/compiler.ts

```diff
@@ -4617,79 +4619,216 @@ export class Compiler extends DiagnosticEmitter {
           return this.module.unreachable();
         }
 
-        let targetType = leftType;
-        let instance: Function | null;
-
-        // Mathf.pow if lhs is f32 (result is f32)
-        if (this.currentType.kind == TypeKind.F32) {
-          rightExpr = this.compileExpression(right, Type.f32, Constraints.CONV_IMPLICIT);
+        let isLeftTypeBool = leftType.kind == TypeKind.BOOL;
```

Comment:
This seems too uncommon to me to make a special case of it. I'd expect that `**` with bools behaves like

* leftType and rightType is bool/bool -> remain bool and wrap analogous to i8/i8
* not both bool -> use the other type due to commonDenominator computation
* compound with leftType bool and rightType something else -> implicit conversion error from i32 to bool down the road

None of this should require special code and be instead implied by general wrapping and conversion rules.

Reply:
alright. I removed special case for booleans: https://github.com/AssemblyScript/assemblyscript/pull/1146/commits/1acd61ce6f52bb490d9b9d499b4de66f4441fca5